src/components/homepage.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate, useLocation } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { FaSearchLocation } from 'react-icons/fa';
+import FetchStats from '../redux/data';
+import { GetStats } from '../redux/stats';
+
+const HomePage = () => {
+  const countryStore = useSelector((store) => store.details);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (countryStore.length === 0) {
+      FetchStats()
+        .then((response) => dispatch(GetStats(response)));
+    }
+  }, []);
+
+  let Africa = countryStore.filter((item) => item.continent === 'Africa');
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const search = query.get('search') || '';
+  Africa = Africa.filter((country) => country.country.includes(search.toLowerCase()));
+  const navigate = useNavigate();
+  const [searchValue, setSearchValue] = useState(search);
+
+  const countryFilterOnChange = (event) => {
+    navigate(event.target.value ? `?search=${event.target.value}` : '');
+    setSearchValue(event.target.value);
+  };
+
+  return (
+    <div className="homePage">
+      <h1 className="continent">Africa</h1>
+      <form className="formSearch">
+        <div>
+          <FaSearchLocation />
+        </div>
+        <div>
+          <input className="searchInput" type="text" value={searchValue} placeholder="Search for country..." onChange={countryFilterOnChange} />
+        </div>
+      </form>
+      <ul className="dataList">
+        {
```

Comment:
- [ ] Great using a Filter stateless component to filter data retrieved from the API 👏🏻 However it does not work fine
We use the .filter() method on an array of countries so that we only return the country that matches a particular country code or uses given parameters. kindly make this work well 💯 👍🏻 

![28capstone](https://user-images.githubusercontent.com/79058364/176301588-b0b6b432-e2bf-4461-8656-b7b324f2b1b3.png)

Reply:
Can you help me in this please?

Reply:
Can you help me in this please?

Reply:
Can you help me with this, please?

Reply:
@ Hello Fatima!!
Sorry for the late reply, let me know where you are with the changes?

Reply:
> * Great using a Filter stateless component to filter data retrieved from the API 👏🏻 However it does not work fine
>   We use the .filter() method on an array of countries so that we only return the country that matches a particular country code or uses given parameters. kindly make this work well 💯 👍🏻

Can you help me plz?